website/content/docs/upgrading/instructions/testing-experimental-raft-backend.mdx

```diff
@@ -0,0 +1,420 @@
+most efficiently.
+
+Other reliability issues such as [followers being unable to catch
+up](/consul/docs/agent/telemetry#raft-replication-capacity-issues) also stem
+from this need to carefully balance the size of the BoltDB log store against how
+long snapshots take to restore - there is a simple solution to that issue if
+letting logs grow much larger to ensure recovery didn't have a potentially
+catastrophic impact on the cluster's write performance.
+
+While not every user will experience a huge difference in performance, the WAL
+backend avoids these performance concerns entirely. It is more performant when
+directly measured due to solving a simpler storage problem than BoltDB was
+designed for. For example it can commit a single log entry with on fsync instead
+of two, and tends to write 2-3x fewer bytes to the disk to do it. The real
+benefit though is that retaining more logs won't impact write performance at all
+and so reducing disk IO with slower snapshots or keeping logs around to catch up
+slower followers are all possible.
+
+## Why try it out?
+
+The new WAL backend has been tested thoroughly during development:
+ * Every component in the WAL like [metadata
+   management](https://github.com/hashicorp/raft-wal/blob/main/types/meta.go),
+   [log file
+   encoding](https://github.com/hashicorp/raft-wal/blob/main/types/segment.go)
+   to actual [file-system
+   interaction](https://github.com/hashicorp/raft-wal/blob/main/types/vfs.go)
+   was abstracted so unit tests can simulate all sorts of difficult-to-reproduce
+   disk failures.
+ * We [used
+   ALICE](https://github.com/hashicorp/raft-wal/blob/main/alice/README.md), to
+   exhaustively simulate thousands of possible crash failure scenarios and test
+   that WAL correctly recovered from each.
+ * We ran hundreds of tests over a few weeks in a performance testing cluster
+   with checksum verification enabled and detected zero cases of data loss or
+   corruption. We plan to continue testing this continuously over the next few
+   months too before making it the default backend.
+
+However, we are well aware of both how complex and how critical disk-persistence
+is for our user's data.
+
+Our hope is that we will have many users at all degrees of scale try WAL in
+their environments after upgrading to 1.15 and report success or failure back so
+we have increased confidence before we make it the default for new clusters.
+
+This guide describes how to safely try and verify it without risking the
+availability of your cluster should there be a latent data-loss issue
+discovered.
+
+## Requirements
+
+- All servers in the Datacenter should be upgraded to Consul 1.15 using the
+  [standard upgrade procedure](/consul/docs/upgrading/general-process) and
+  the [1.15 upgrade notes](/consul/docs/upgrading/upgrade-specific#consul-1-15-x).
+- You need a Consul cluster with at least 3 nodes to safely test the new
```

Comment:
Is this enough?

We mention below that enabling WAL on a voting server might itself generate an issue:

> If you enable a server to use WAL using Consul OSS or on a voting server with
   Consul Enterprise, it's _possible_ that the WAL could cause corruption of
   that server's state (with the caveats above) _and then_ become the leader and
   replicate that corruption to all other servers.

Or is this more related to the fact then even in case it goes all good users with one server will have downtime?

To be safe the real requirement is to be able to use non-voting servers aka to use Consul Enterprise